autospotting.go

```diff
@@ -4,218 +4,163 @@ import (
-		conf.SpotPriceBufferPercentage,
-		conf.BiddingPolicy,
-		conf.FilterByTags,
-		conf.TagFilteringMode,
-		conf.SpotProductDescription,
-		conf.InstanceTerminationMethod,
-		conf.TerminationNotificationAction,
-		conf.CronSchedule,
-		conf.CronScheduleState,
-		conf.LicenseType,
-	)
-
-	autospotting.Run(conf.Config)
+	as.EventHandler(event)
 	log.Println("Execution completed, nothing left to do")
 }
 
 // this is the equivalent of a main for when running from Lambda, but on Lambda
-// the run() is executed within the handler function every time we have an event
+// the runFromCronEvent() is executed within the handler function every time we have an event
 func init() {
+
 	var region string
 
+	as = &autospotting.AutoSpotting{}
+
 	if r := os.Getenv("AWS_REGION"); r != "" {
 		region = r
 	} else {
 		region = endpoints.UsEast1RegionID
 	}
 
-	conf = &cfgData{
-		&autospotting.Config{
-			LogFile:         os.Stdout,
-			LogFlag:         log.Ldate | log.Ltime | log.Lshortfile,
-			MainRegion:      region,
-			SleepMultiplier: 1,
-			Version:         Version,
-			LicenseType:     os.Getenv("LICENSE"),
-		},
+	conf = autospotting.Config{
+		LogFile:         os.Stdout,
+		LogFlag:         log.Ldate | log.Ltime | log.Lshortfile,
+		MainRegion:      region,
+		SleepMultiplier: 1,
+		Version:         Version,
+		LicenseType:     os.Getenv("LICENSE"),
+		LambdaManageASG: os.Getenv("LAMBDA_MANAGE_ASG"),
 	}
+	parseCommandLineFlags()
 
-	conf.initialize()
-
+	as.Init(&conf)
 }
 
-// Handler implements the AWS Lambda handler
+// Handler implements the AWS Lambda handler interface
 func Handler(ctx context.Context, rawEvent json.RawMessage) {
-
-	var snsEvent events.SNSEvent
-	var cloudwatchEvent events.CloudWatchEvent
-	parseEvent := rawEvent
-
-	// Try to parse event as an Sns Message
-	if err := json.Unmarshal(parseEvent, &snsEvent); err != nil {
-		log.Println(err.Error())
-		return
-	}
-
-	// If event is from Sns - extract Cloudwatch's one
-	if snsEvent.Records != nil {
-		snsRecord := snsEvent.Records[0]
-		parseEvent = []byte(snsRecord.SNS.Message)
-	}
-
-	// Try to parse event as Cloudwatch Event Rule
-	if err := json.Unmarshal(parseEvent, &cloudwatchEvent); err != nil {
-		log.Println(err.Error())
-		return
-	}
-
-	// If event is Instance Spot Interruption
-	if cloudwatchEvent.DetailType == "EC2 Spot Instance Interruption Warning" {
-		if instanceID, err := autospotting.GetInstanceIDDueForTermination(cloudwatchEvent); err != nil {
-			return
-		} else if instanceID != nil {
-			spotTermination := autospotting.NewSpotTermination(cloudwatchEvent.Region)
-			spotTermination.ExecuteAction(instanceID, conf.TerminationNotificationAction)
-		}
-	} else {
-		// Event is Autospotting Cron Scheduling
-		run()
-	}
+	eventHandler(&rawEvent)
 }
 
-// Configuration handling
-func (c *cfgData) initialize() {
-
-	c.parseCommandLineFlags()
-
-	data, err := ec2instancesinfo.Data()
-	if err != nil {
-		log.Fatal(err.Error())
-	}
-	c.InstanceData = data
```

Comment:
@cristim I don't see where this is populated anymore. Is it no longer necessary with these changes?

Reply:
Looks like this logic was moved to core/main.go around line 40.

Reply:
Ah I see. My mistake. Ctrl-F turned up nothing but that's because GitHub collapsed that file 🤦‍♂